corehq/project_limits/models.py

```diff
@@ -97,3 +99,21 @@ class Meta:
     def __str__(self):
         domain = f"[{self.domain}] " if self.domain else ""
         return f"{domain}{self.key}: {self.limit}"
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        self.for_key.clear(self.__class__, self.key, self.domain)
+
+    @classmethod
+    @quickcache(['key', 'domain'], timeout=7 * 24 * 60 * 60)
+    def for_key(cls, key, domain=''):
+        """
+        Return the value associated with the given key, prioritizing the domain specific entry over the general one
+        The timeout is long because this is a small table that is effectively storing key/value pairs in redis
+        """
+        domain_filter = models.Q(domain="")
+        if domain:
+            domain_filter |= models.Q(domain=domain)
+        filters = models.Q(key=key) & domain_filter
+        limit = SystemLimit.objects.filter(filters).order_by("-domain").values_list("limit", flat=True).first()
+        return None if limit is None else limit  # limit could be 0
```

Comment:
nit: How is this different from `return limit`?
```suggestion
        return SystemLimit.objects.filter(filters).order_by("-domain").values_list("limit", flat=True).first()
```

Reply:
[11c4e63](https://github.com/dimagi/commcare-hq/pull/35781/commits/11c4e63bb2fee263f47da3ec7726af544d2be5f2)